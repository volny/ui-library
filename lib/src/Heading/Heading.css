:root {
  --font-size-1: 39px;
  --font-size-2: 31.2px;
  --font-size-3: 20px;
  --font-size-4: 16px;
  --font-size-5: 12.8px;
}

.Heading {
  composes: sansSerif fontWeightNormal antialiased from "../typography.css";
  margin-bottom: 0;
  margin-top: 0;
}

.fontSize1 {
  composes: darkGrey from "../colors.css";
  font-size: var(--font-size-1);
}

.fontSize2 {
  composes: darkGrey from "../colors.css";
  font-size: var(--font-size-2);
}

.fontSize3 {
  composes: darkGrey from "../colors.css";
  font-size: var(--font-size-3);
}

.fontSize4 {
  composes: darkGrey70 from "../colors.css";
  font-size: var(--font-size-4);
}

.fontSize5 {
  composes: darkGrey70 from "../colors.css";
  font-size: var(--font-size-5);
}
