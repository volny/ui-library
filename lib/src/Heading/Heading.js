import * as React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

import styles from './Heading.css'
import colors from '../colors.css'

const Heading = ({ children, color, level }) => {
  const cs = cx(styles.Heading, styles[`fontSize${level}`], colors[color])

  return React.createElement(
    `h${level}`,
    {
      className: cs,
    },
    children
  )
}

Heading.defaultProps = {
  level: '1',
}

Heading.propTypes = {
  children: PropTypes.node.isRequired,
  color: PropTypes.oneOf([
    'darkGrey',
    'white',
    'green',
    'darkGrey70',
    'darkGrey30',
    'darkGrey10',
    'white90',
    'white30',
    'green30',
    'green20',
    'yellow',
    'red',
  ]),
  level: PropTypes.string,
}

export default Heading
