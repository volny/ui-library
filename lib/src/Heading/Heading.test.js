/**
 * @jest-environment node
 */

import React from 'react'
import { create } from 'react-test-renderer'
import Heading from './Heading'

test('Heading h1', () => {
  const tree = create(<Heading>h</Heading>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Heading h2', () => {
  const tree = create(<Heading level="2">h</Heading>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Heading with color and level 3', () => {
  const tree = create(<Heading level="3" color="yellow">h</Heading>).toJSON()
  expect(tree).toMatchSnapshot()
})
