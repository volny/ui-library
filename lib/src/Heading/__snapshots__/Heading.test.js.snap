// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading h1 1`] = `
<h1
  className="Heading fontSize1 undefined"
>
  h
</h1>
`;

exports[`Heading h2 1`] = `
<h2
  className="Heading fontSize2 undefined"
>
  h
</h2>
`;

exports[`Heading with color and level 3 1`] = `
<h3
  className="Heading fontSize3 yellow"
>
  h
</h3>
`;
